### PR TITLE
Nano: Add key feature for keras inference 

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add key feature for keras inference

### 4. How to test?
- [ ] Document test
       https://yishuotestdoc.readthedocs.io/en/add-keras-optimizer-doc/doc/Nano/Overview/tensorflow_inference.html

### CODE REVIEW

**Comment by MeouSker77** [comment]
@ACupofAir @Siritao please have a look as well. Please feel free to raise any comments

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 1]
```diff
@@ -1,98 +1,208 @@
...
-## Quantization
-Quantization is widely used to compress models to a lower precision, which not only reduces the model size but also accelerates inference. BigDL-Nano provides `Model.quantize()` API for users to quickly obtain a quantized model with accuracy control by specifying a few arguments. `Sequential` has similar usage, so we will only show how to use an instance of `Model` to enable quantization pipeline here.
+```eval_rst
+.. note::
+    You can install all required dependencies by
+
+    ::
 
-To use INC as your quantization engine, you can choose accelerator as `None` or `'onnxruntime'`. Otherwise, `accelerator='openvino'` means using OpenVINO POT to do quantization.
+        pip install bigdl-nano[tensorflow,inference]
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:31]: shall we add `--pre --upgrade` here ?

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 2]
```diff
@@ -1,98 +1,208 @@
...
+
+Still taking the example in [Runtime Acceleration](#runtime-acceleration), you can use it as following:
+
+```python
+# try all supproted optimizations
+opt = InferenceOptimizer()
+opt.optimize(model, x=train_dataset)
+
+# get the best optimization
+best_model = opt.get_best_model()
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:191]: `best_model, option = opt.get_best_model()`
the output of `get_best_model` is a tuple

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 3]
```diff
@@ -1,98 +1,208 @@
...
+
+# use the quantized model as before
+y_hat = best_model(train_examples)
+best_model.predict(train_dataset)
+```
+
+`InferenceOptimizer.optimize` will try all supported optimizations and choose the best one. It also uses parameter `x` and `y` to receive calibration data like `InferenceOptimizer.quantize`.
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:198]: Maybe put a output table here for better display effect?

**Review by rnwang04 — COMMENTED** [review]
others LGTM.

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md (hunk 1 above)
Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:31]: we can merge this PR after releasing bigdl-nano 2.2, no need to add `--pre --upgrade`

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 4]
```diff
@@ -1,98 +1,225 @@
...
+traced_model.predict(train_dataset)
+```
+
+## Quantization
+
+Quantization is widely used to compress models to a lower precision, which not only reduces the model size but also accelerates inference. BigDL-Nano provides `InferenceOptimizer.quantize()` API for users to quickly obtain a quantized model with accuracy control by specifying a few arguments. `Sequential` has similar usage, so we will only show how to use an instance of `Model` to enable quantization pipeline here.
+
+To use INC as your quantization engine, you can choose `accelerator=None/'onnxruntime'`. Otherwise, `accelerator='openvino'` means using OpenVINO POT (Post-training Optimization) to do quantization.
+
+### Quantization without extra accelerator
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:88]: The title here is strange for me. Seems like we are giving example in this section for both with/without extra acceleration?

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 5]
```diff
@@ -1,98 +1,225 @@
...
 q_model.predict(train_dataset)
 ```
+
 This is a most basic usage to quantize a model with defaults, INT8 precision, and without search tuning space to control accuracy drop.
 
-To use quantization, you must use functional API to create a keras model. This is a known limitation
-of INC.
+```eval_rst
+.. note::
+    To use quantization, you must use functional API to create a keras model. This is a known limitation of INC.
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:116]: Does it mean here that for quantization using INC, we can only use `keras.Model` instead of `keras.Sequential`? Does that mean we could still use POT quantization for `keras.Sequential`?

If so, maybe we should refactor this sentence a little bit, so that we could emphasize that `keras.Sequential` can still be used for POT quantization.

**Review by Oscilloscope98 — COMMENTED** [review]
Others LGTM :)

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md (hunk 4 above)
Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:88]: this is a typo, it should be "Quantization without Accuracy Control"

File: docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md [hunk 6]
```diff
@@ -1,98 +1,225 @@
...
+All available runtime accelerations are integrated in `InferenceOptimizer.trace(accelerator='onnxruntime'/'openvino')` with different accelerator values.
+
+Taking MobileNetV2 as an example, you can use runtime acceleration as below:
+
 ```python
 import tensorflow as tf
 from tensorflow.keras.applications.mobilenet_v2 import MobileNetV2
 import numpy as np
-from bigdl.nano.tf.keras import Model
+from bigdl.nano.tf.keras import Model, InferenceOptimizer
```
[the remark this answers is not in the document]
Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md:54]: I have tried it, we don't have to create model use `bigdl.nano.tf.keras.Model` now, I'll change it
